python/lsst/ip/diffim/imageMapReduce.py

```diff
@@ -275,15 +276,36 @@ def run(self, mapperResults, exposure, **kwargs):
                     wtsView = afwImage.ImageI(weights, item.getBBox())
                     wtsView.getArray()[isNotNan] += 1
 
+        isNan = (np.isnan(newMI.getImage().getArray()) |
+                  np.isnan(newMI.getVariance().getArray()))
+        if np.sum(isNan) > 0:
+            # set mask to INVALID for pixels where produced exposure is NaN
+            mask = newMI.getMask()
+            # New mask plane - for debugging map-reduced images
+            mask.addMaskPlane('INVALID_MAPREDUCE')
+            bad = mask.getPlaneBitMask(['INVALID_MAPREDUCE', 'BAD', 'NO_DATA'])
+            mask.getArray()[isNan] |= bad
+
         if reduceOp == 'average':
             wts = weights.getArray().astype(np.float)
             self.log.info('AVERAGE: Maximum overlap: %f', wts.max())
             self.log.info('AVERAGE: Average overlap: %f', np.nanmean(wts))
+            self.log.info('AVERAGE: Minimum overlap: %f', wts.min())
+            wtsZero = wts == 0.
+            self.log.info('AVERAGE: Number of zero pixels: %f (%f%%)', np.sum(wtsZero),
+                          np.mean(wtsZero) * 100.)
+            wts[wtsZero] = 1e-29  # avoid division by zero warnings -- will set to nan and flag them below
             newMI.getImage().getArray()[:, :] /= wts
             newMI.getVariance().getArray()[:, :] /= wts
```

Comment:
numpy ufuncs save us here, you can just do:
np.divide(newMI.getImage().getArray(), wts, out=newMI.getImage().getArray(), where=~wtsZero)
This will do inplace operation (output is the same as input) and only loop over and do a calculation where wtsZero is not true, all in one step.